pkg/agent/util/winnet/types.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package winnet
+
+import (
+	"fmt"
+	"net"
+
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	iputil "antrea.io/antrea/pkg/util/ip"
+)
+
+type Route struct {
+	LinkIndex         int
+	DestinationSubnet *net.IPNet
+	GatewayAddress    net.IP
+	RouteMetric       int
+}
+
+type Neighbor struct {
+	LinkIndex        int
+	IPAddress        net.IP
+	LinkLayerAddress net.HardwareAddr
+	State            string
+}
+
+type NetNatStaticMapping struct {
+	Name         string
+	ExternalIP   net.IP
+	ExternalPort uint16
+	InternalIP   net.IP
+	InternalPort uint16
+	Protocol     binding.Protocol
+}
+
+func (r Route) String() string {
```

Comment:
This function used to have a pointer receiver (`r *Route`). Was there a specific reason for changing to a pointer receiver?

There are a few other methods below that also use a value receiver, and I am not sure why.

The structs are not really large so maybe it doesn't matter, but I wonder if it was a conscious decision?

Reply:
They should have a pointer receiver. I didn't catch it before. Thanks for reminding it.